core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -235,6 +235,23 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
           }
 
           tryBuilder.map(_.result())
+        case o: ConfigObject =>
```

Comment:
The two signatures are similar in this case but, practically speaking, if you use `Nothing` then you don't have to specify what `U` is and in the context of `reportBadIndex` we don't need to specify it. In my opinion this makes sense to use `Nothing` because `reportBadIndex` shouldn't care about what a successful values would be.